### DISCUSSION TITLE
v6 upg help need

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#638] [Q&A (Help)] [answered]
Doing Step 20 on one my project i've this error:
<img width="1422" alt="bkp6-upg-hint-err" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/63a59893-d13a-42dd-bd5b-ffe4bf495b6e">

Some help to check path setting for "backpack::" ?

Thanks

**Comment by blondie63** [comment]
@tabacitu I see the problem is with the new namespace, I looked at the backpack demo and saw that at resources/views/vendor/backpack/ui/inc there is only the menu_items.blade.php file and the blades are at resources/views/ vendor/backpack/theme-tabler/inc

I tried to move my blades there but it can't find them!
Could someone please clarify the new way to manage the flie blades to include, the ones that in BKP5 were in resources/views/vendor/backpack/base/inc

Thank you

**Reply by pxpm** [reply]
If you are using a theme and the theme overwrites that file, it should go inside `resources/vendor/backpack/theme-XXX/....`, in case it's a file that the theme does not overwrite is gotten from the UI so you should use `resources/vendor/backpack/ui/...`. 

Then you have now the `crud::` namespace and the `backpack.ui::`. A good place to view how we use them is in the helper function https://github.com/Laravel-Backpack/CRUD/blob/96ac869f152a763a6956c10f9f7e1e164e613a4b/src/helpers.php#L246

You can use `backpack_view('the_view_you_want')` and we will search in the locations mentioned in the helper link I sent you. 

A little bit more info here: https://backpackforlaravel.com/docs/6.x/base-themes#how-themes-work-1

Let me know if that does help. 

Cheers

**Comment by blondie63** [comment]
Hi @pxpm finally i've converted my app and now start, i've only to convert to use basset..
do you know how to convert this row for basset?
<script src="{{ asset('js/fetch-company.js') }}" defer></script>

**Comment by blondie63** [comment]
I also have another problem, one of my developers had made some customizations that you see below:
<img width="424" alt="pic-bkp6-theme" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/cfb26679-55ae-4516-8771-47549c5e5392">

Original theme-tabler are here:
<img width="426" alt="pic-bkp6-theme-original" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/bc9f40a6-b4c0-4a04-a6d4-ccb56df601a9">

Now the theme-table structure is managed by files with different names, is there a map to understand their use?
For example, "BACKPACK" now appears at the top left, while before I had my logo and the version of the app I managed in the inc/main_header.blade.php file, which file should I now edit?

BKP5
<img width="586" alt="pic-BKP5" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/29281f56-0106-4a6e-a715-89ded2520c38">

BKP6
<img width="705" alt="pic-BKP6" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/157d9a7a-92d9-4322-9371-9d9dac7b48ff">

Thanks for help

**Reply by pxpm** [reply]
Hey @blondie63 it now depends on theme, and in what layout you use. 

I would assume we are talking Tabler vertical. https://github.com/Laravel-Backpack/theme-tabler/blob/fad0d3cbe045178e8e52fb2ac635f1c5e69c48ea/resources/views/layouts/_vertical/menu_container.blade.php#L7

Tabler will include the same file name, but since now themes take precedence over ui components, you are getting the theme files (defaults) instead of your custom ones. You need to place the customized files in **resources/views/vendor/backpack/theme-tabler/** `inc/topbar_left_content.blade.php` for example. 

Let me know how it goes.

**Reply by blondie63** [reply]
Thanks @pxpm I see that these days you are often updating backpack and today also tabler, how can I make my customizations while maintaining Tabler updates as well?

**Reply by pxpm** [reply]
Hey @blondie63 hope to find you well. 
It all boils down to what type of costumizations, most of the time we can think in some solution in crud or even in the theme to make less overwrites. 

What I always try to do is to include the backpack vendor file in my custom file, and do my customizations on top of it, without overwritting the actual core files. 
It's not possible everytime, so a good thing is to subscribe to repository and keep the files updated. No way around it. 

Maybe a good tool that we could build is something like a command that identifies the files you have overwritten and when you do a `composer update` or similar it will warn you if some of the overriden files have changed.

This may be a good solution to help developers to keep up with the changes. 

It's usual to have more changes in core files after the release of a major version, but we hope to make them stable and configurable enough that developers don't have to overwrite them for small stuff. 

Food for tought. I will bring this idea in our next team meeting and see the feedback 🙏 

Cheers

**Reply by tabacitu** [reply]
@blondie63 my 2 cents... with that many customizations... you're no longer using the Tabler theme, basically. Create your own Backpack theme... then you know everything regarding the interface is in your control. Then you KNOW you're not getting any Tabler updates, because you have your own theme (even if it's based on Tabler). It'll be easier to work that way.

**Comment by blondie63** [comment]
@tabacitu I wanna show you my customizations done on BKP5 CoreUI:
<img width="649" alt="pic-bkp5-login" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/b7b88967-9c93-4c01-97ef-af2b6677f70d">
<img width="1138" alt="pic-bkp5-sidebar-menu-company-admin" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/da27924a-a59c-4e09-a642-01149b971e0d">
<img width="1079" alt="pic-bkp5-sidebar-menu-supervisor" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/766cacf9-e322-4bfb-b544-815c14c1bc2c">

What you think ?

**Reply by tabacitu** [reply, marked as answer]
Looks like you've customized the login page, the register page, the sidebar, the footer. So... quite a bit. I would create a different theme, yes. That way you are 100% in control of what's happening. 

OR you can add just the files you need in your theme, and use the `view_namespace_fallback` in `config/backpack/ui.php` to point to the theme you based it on. That way you don't have a lot of files to maintain. But... you're tied to an existing theme...

Your choice. I just said what I would do.
